content/autocon1.md

```diff
@@ -0,0 +1,60 @@
+
+Want to see a real-life use case? I recommend watching the video ["How We Are Moving from GitOps to Kubernetes Resource Model in 5G Core"](https://www.youtube.com/watch?v=crmTnB6Zwt8) from the last KubeCon.
+
+
+## Standards
+
+Standards and YANG models were a hot topic at AutoCon. It's important to avoid mixing different types of standards. For example, when discussing models/schemas, we shouldn't bring protocols into the discussion. So, when talking about OpenConfig models, we should not mix in points regarding NetConf or gRPC. Each layer should be addressed separately.
+We have and need standards for communicating with devices (e.g., CLI over Telnet or SSH, SNMP over UDP or TCP, NetConf RPCs over SSH or gRPC). At the encoding level, we also have a variety of options, though modern interfaces often use verbose formats like XML and JSON or binary-encoded options like ProtoBuf or CBOR.
+
+At the schema layer, where we describe models and structure the data we send or receive, we need two distinct discussions: one about the language itself and the other about the defined models. Schema languages include XSD (widely used for SOAP), YANG (inspired by XSD and used for NetConf, RestConf, and gRPC for network devices), and JSONSchema (used by OpenAPI 3.1 and commonly in REST APIs, especially when interacting with third-party systems).
+For me, the key is having a comfortable way to work with data schema languages within a robust ecosystem. Unfortunately, YANG has advanced features that make auto-generating SDKs and working with it difficult. Unless the industry demands better solutions, we won't see much change. Remember, the hardware we install now will likely run for a long time.
+Regarding standardised models for device communication, we need to be realistic. SNMP, CLI, OpenConfig, and IETF YANG models have shown that it's not practical to have one model that rules them all. Even different platforms or NOS versions will have slight deviations in the models, necessitating a translation layer. Therefore, we should focus on making schema handling more user-friendly.
+One point to consider: Why is YANG not easily consumable? Why do so many render the XML payload for NetConf with jinja2 and treat it like a text file? Is this really the preferred method?
+
+
+On the abstraction layer, where we model the intent of the network, a unified approach seems more feasible. We could translate and normalise data from device APIs into abstract models. The IETF is working on many YANG models, such as in the areas of Digital Map and Digital Twin. For more details, check out this IETF draft.
```

Comment:
I think you're missing a link here for the IETF draft :)

Reply:
Thanks. yes I forgot to add the link here :o

Reply:
Thanks. yes I forgot to add the link here :o